tokio/src/task/join_set.rs

```diff
@@ -535,7 +535,9 @@ impl<T: 'static> JoinSet<T> {
     ///  * `Poll::Ready(None)` if the `JoinSet` is empty.
     ///
     /// Note that this method may return `Poll::Pending` even if one of the tasks has completed.
-    /// This can happen if the [coop budget] is reached.
+    /// This can happen if the [coop budget] is reached. However, the task's `JoinHandle` is only
+    /// consumed (and the task ID freed for potential re-use) if this method returns
+    /// `Poll::Ready(Some(_))`.
```

Comment:
I prefer a dedicated section instead of writing the same explanation for each method.